src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -934,6 +941,12 @@ internal async Task SearchPackagesAndRefreshUpdateCountAsync(string searchText,
                 {
                     await RefreshInstalledAndUpdatesTabsAsync();
                 }
+                else
+                {
+                    // If warning icon is still visible, show the vulnerabilities filter.
+                    _topPanel.CheckBoxVulnerabilities.IsEnabled = _topPanel._warningIcon.IsVisible ? true : false;
```

Comment:
I challenge whether this is useful behavior. Disabling the filter just because there aren't items that meet the criteria may not be as beneficial as it sounds. It _could_ be confusing if customers want to leave it on and do multiple searches (or change other filters).

My vote is we leave the filter enabled all the time. Then this bug isn't relevant: https://github.com/NuGet/Home/issues/12868

Reply:
The problem with leaving it enabled is that users can use it when they are no vulnerabilities information.

Let say a user checks it and see no vulnerabilities, the user won't know that vulnerabilities data is still being fetch. The user, thinking that there are vulnerabilities, unchecks the checkbox before the data is available, in this scenario they will never know that he has vulnerabilities he can address in the PM UI.

Reply:
I agree that's a problem, but I disagree that the solution is disabling the checkbox. The solution would be to ensure a loading indicator remains active until vulnerability data is retrieved. 
https://github.com/NuGet/Home/issues/10811

Reply:
+1 to Donnie's suggestion :)